server/app/lib/utils/cloud_forms/provider.rb

```diff
@@ -64,6 +64,32 @@ def self.add(cfme_ip, provider_params)
         }
         request_data = new_provider_form.request_data
         provider_added = agent.post("https://#{cfme_ip}/ems_infra/create/new?button=add", request_data, submit_headers)
+
+        host_list = agent.get("https://#{cfme_ip}/host/show_list", [], agent.page.uri)
```

Comment:
The provider 'add' method is getting a bit long.  Would it make sense to break this logic out in to a separate method (or methods) invoked by the 'add'?

(You do not have to do this; however, it is a suggestion for the long-term for readability, testability...etc)